TicTacToe.Foundation/Games/Game.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using TicTacToe.Foundation.Boards;
+using TicTacToe.Foundation.Interfaces;
+using TicTacToe.Foundation.Games.StepResults;
+using TicTacToe.Foundation.Games.GameResults;
+
+namespace TicTacToe.Foundation.Games
+{
+    public class Game : IGame
+    {
+        private readonly IIoProvider _ioProvider;
+        private readonly IReadOnlyCollection<IPlayer> _players;
+        private readonly IBoardInternal _board;
+        private readonly IReadOnlyCollection<IWinningState> _winningStates;
+
+        private int _currentPlayerIndex;
```

Comment:
Avoid duplicates, should be `single source of truth`, calculate Player from index.